src/pynxtools/nomad/schema.py

```diff
@@ -137,6 +138,74 @@ class NexusActivityStep(ActivityStep):
     )
 
 
+class NexusIdentifier(EntityReference):
+    name = Quantity(
+        type=str,
+        description="The name of the nexus identifier field.",
+    )
+
+    def normalize(self, archive, logger):
+        from nomad.processing import Entry
+
+        def create_Entity(lab_id, archive, f_name, qunt_name):
+            entitySec = Entity()
+            entitySec.lab_id = lab_id
+            entitySec.name = qunt_name
+            entity = EntryArchive(
+                data=entitySec,
+                m_context=archive.m_context,
+                metadata=EntryMetadata(
+                    entry_type="identifier", domain="nexus", readonly=True
+                ),  # upload_id=archive.m_context.upload_id,
+            )
+            with archive.m_context.raw_file(f_name, "w") as f_obj:
+                json.dump(entity.m_to_dict(with_meta=True), f_obj)
+            archive.m_context.process_updated_raw_file(f_name)
+
+        def get_entry_reference(archive, f_name):
+            """Returns a reference to data from entry."""
+            from nomad.utils import hash
+
+            upload_id = archive.metadata.upload_id
+            entry_id = hash(upload_id, f_name)
+            return f"../uploads/{upload_id}/archive/{entry_id}#data"
+
+        super().normalize(archive, logger)
+        if not self.reference:
+            lab_hash = hashlib.md5(self.lab_id.encode()).hexdigest()
+
+            f_name = f"{self.name}_{lab_hash}.archive.json"
```

Comment:
class_name could be used as prefix. Is `name` always valid for filename, too?

Reply:
The `name` is constructed from class_name and field_name field to infer path of identifier. 
I think, `name` should be always valid if an identifierNAME is initialized, otherwise it would not initialize the NexusIdentifier.